app/services/search/google_search_service.rb

```diff
@@ -13,9 +13,9 @@ def search
 
       response = Faraday.get(uri)
 
-      # TODO: Handle error in background job
-      # https://github.com/nimblehq/ic-rails-huey-toby/issues/19
       response.body if response.status == 200
+    rescue Faraday::ConnectionFailed
+      raise IcRailsHueyToby::Errors::SearchServiceError
```

Comment:
Just curious why don't we raise a more specific error message, i.e: "UnSupportedSearchEngine"? For security reason? 🤔

Reply:
Originally I raised `Faraday::ConnectionFailed`, but this approach simplifies the code since the action we take is the same for all errors anyways (marking the search result as failed) 🙏 

**Reference**: https://github.com/nimblehq/ic-rails-huey-toby/pull/33#discussion_r1227471852